File: src/main/resources/static/src/table_settings/letter-pack.js
export default {
    colHeaders: function(index) {
        let self = this;
        return self.columns[index].value;
    },
    columns: [{
            data: 'toPost',
            value: '[To]郵便番号',
            validator: /^\d{3}[-]\d{4}$|^\d{7}$/,
            comment: '半角数字7桁で入力してください\n\n[To]おところが空欄の場合、\n郵便番号入力後に[To]おところを自動入力します。',
        },
        {
            data: 'toAddres',
            value: '[To]おところ',
        },
        {
            data: 'toName',
            value: '[To]おなまえ',
        },
        {
            data: 'toTel',
            value: '[To]電話番号',
            validator: /0(?:\d[(-]?\d{4}|(?![789]0)\d{2}[(-]?\d{3}|\d{3}[(-]?\d{2}|\d{4}[(-]?\d|[789]0[(-]?\d{4})[)-]?\d{4}(?!\d)|0/g,
            comment: '半角数字で入力してください。\n\n届け先の電話番号が不明の場合は\n半角数字で　0　を入力してください\n仕上がったPDFでは空欄になります。',
        },
        {
            data: 'fromPost',
            value: '[From]郵便番号',
            validator: /^\d{3}[-]\d{4}$|^\d{7}$/,
            comment: '半角数字7桁で入力してください\n\n[From]おところが空欄の場合、\n郵便番号入力後に[From]おところを自動入力します。',
        },
        {
            data: 'fromAddres',
            value: '[From]おところ',
        },
        {
            data: 'fromName',
            value: '[From]おなまえ',
        },
        {
            data: 'fromTel',
            value: '[From]電話番号',
            validator: /0(?:\d[(-]?\d{4}|(?![789]0)\d{2}[(-]?\d{3}|\d{3}[(-]?\d{2}|\d{4}[(-]?\d|[789]0[(-]?\d{4})[)-]?\d{4}(?!\d)/g,
            comment: '半角数字で入力してください。\n\nご依頼主の電話番号を空欄にすることはできません。',
        },
        {
            data: 'itemName',
            value: '品名'
        }
    ]
}

File: src/main/java/com/hand/dot/data/templates/LetterPack.java
package com.hand.dot.data.templates;

import com.hand.dot.util.StringUtil;

public class LetterPack extends PdfTemplate{

	//[届け先]郵便番号
	private String toPost;
	//[届け先]おところ
	private String toAddres;
	//[届け先]おなまえ
	private String toName;
	//[届け先]電話番号
	private String toTel;
	//[依頼主]郵便番号
	private String fromPost;
	//[依頼主]おところ
	private String fromAddres;
	//[依頼主]おなまえ
	private String fromName;
	//[依頼主]電話番号
	private String fromTel;
	//品名
	private String itemName;

	public String getToPost() {
		return toPost;
	}
	public void setToPost(String toPost) {
		this.toPost = StringUtil.appendWithSeparators(StringUtil.removeHyphen(toPost));
	}
	public String getToAddres() {
		return toAddres;
	}
	public void setToAddres(String toAddres) {
		this.toAddres = toAddres;
	}
	public String getToName() {
		return toName;
	}
	public void setToName(String toName) {
		this.toName = toName;
	}
	public String getToTel() {
		return toTel;
	}
	public void setToTel(String toTel) {
		if(toTel != null && toTel.equals("0")){
			this.toTel = "";
			return;
		}
		this.toTel = toTel;
	}
	public String getFromPost() {
		return fromPost;
	}
	public void setFromPost(String fromPost) {
		this.fromPost = StringUtil.appendWithSeparators(StringUtil.removeHyphen(fromPost));
	}
	public String getFromAddres() {
		return fromAddres;
	}
	public void setFromAddres(String fromAddres) {
		this.fromAddres = fromAddres;
	}
	public String getFromName() {
		return fromName;
	}
	public void setFromName(String fromName) {
		this.fromName = fromName;
	}
	public String getFromTel() {
		return fromTel;
	}
	public void setFromTel(String fromTel) {
		this.fromTel = fromTel;
	}
	public String getItemName() {
		return itemName;
	}
	public void setItemName(String itemName) {
		this.itemName = itemName;
	}

}
